ios/LayoutReanimation/REAAnimationsManager.m

```diff
@@ -288,73 +242,131 @@ - (NSDictionary *)prepareDataForLayoutAnimatingWorklet:(NSMutableDictionary *)cu
-    return;
+  return REANodeFind(view, ^(id<RCTComponent> view) {
+    return [self->_ancestorsOfExitingViews objectForKey:view.reactTag] != nil ||
+        self->_hasAnimationForTag(view.reactTag, @"exiting");
+  });
+}
+
+- (void)registerExitingAncestors:(UIView *)child
+{
+  UIView *parent = child.superview;
+  while (parent != nil && ![parent isKindOfClass:[RCTRootView class]]) {
+    if (parent.reactTag != nil) {
+      _ancestorsOfExitingViews[parent.reactTag] = @([_ancestorsOfExitingViews[parent.reactTag] intValue] + 1);
+    }
+    parent = parent.superview;
   }
-  NSMutableDictionary<NSString *, NSObject *> *startValues = before.values;
-  if (state == Inactive) {
-    if (startValues != nil) {
-      _states[tag] = [NSNumber numberWithInt:ToRemove];
-      [_toRemove addObject:tag];
-      [self scheduleCleaning];
+}
+
+- (void)maybeDropAncestors:(UIView *)child
+{
+  UIView *parent = child.superview;
+  while (parent != nil && ![parent isKindOfClass:[RCTRootView class]]) {
+    UIView *view = parent;
+    parent = parent.superview;
+    if (view.reactTag != nil) {
+      int trackingCount = [_ancestorsOfExitingViews[view.reactTag] intValue] - 1;
+      if (trackingCount <= 0) {
+        if ([_ancestorsToRemove containsObject:view.reactTag]) {
+          [view removeFromSuperview];
+          [_ancestorsToRemove removeObject:view.reactTag];
+        }
+        [_ancestorsOfExitingViews removeObjectForKey:view.reactTag];
+      } else {
+        _ancestorsOfExitingViews[view.reactTag] = @(trackingCount);
+      }
     }
-    return;
   }
-  _states[tag] = [NSNumber numberWithInt:Disappearing];
-  NSDictionary *preparedValues = [self prepareDataForAnimatingWorklet:startValues frameConfig:ExitingFrame];
-  _startAnimationForTag(tag, @"exiting", preparedValues, @(0));
 }
 
-- (void)onViewCreate:(UIView *)view after:(REASnapshot *)after
+- (BOOL)removeRecursive:(UIView *)view fromContainer:(UIView *)container withoutAnimation:(BOOL)shouldRemove;
 {
-  _reaUiManager.flushUiOperations();
-  NSNumber *tag = view.reactTag;
-  if (_states[tag] == nil) {
-    _states[tag] = [NSNumber numberWithInt:Inactive];
-    _viewForTag[tag] = view;
+  if (view.reactTag) {
+    BOOL hasExitAnimation = _hasAnimationForTag(view.reactTag, @"exiting");
+    BOOL wantAnimateExit = hasExitAnimation;
+
+    for (UIView *subview in [view.reactSubviews copy]) {
+      if ([self removeRecursive:subview fromContainer:view withoutAnimation:(shouldRemove && !hasExitAnimation)]) {
+        wantAnimateExit = YES;
+      }
+    }
+
+    if (wantAnimateExit) {
+      REASnapshot *before;
+      if (hasExitAnimation) {
+        before = [[REASnapshot alloc] init:view];
+      }
+      // start exit animation
+      UIView *originalSuperview = view.superview;
+      NSUInteger originalIndex = [originalSuperview.subviews indexOfObjectIdenticalTo:view];
+      [container removeReactSubview:view];
+      view.userInteractionEnabled = NO;
```

Comment:
Maybe let's explain why we disable user interaction here (if there's more to it than just disabling user interaction). If there's nothing more to it than just disabling user interaction, let's also add a comment so future "us" don't wonder if there's more to it than just disabling user interaction.